plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/TestOracleConnectorTest.java

```diff
@@ -83,4 +90,56 @@ protected SqlExecutor onRemoteDatabase()
     {
         return oracleServer::execute;
     }
+
+    @Test
+    public void testPushdownJoinWithLongNameSucceeds()
+    {
+        tryCleanupTemporaryTable();
+        try {
+            String baseColumnName = "test_pushdown_" + randomNameSuffix();
+            String validColumnName = baseColumnName + "z".repeat(MAX_CHARS_COLUMN_ALIAS - baseColumnName.length());
+
+            assertUpdate(format("""
+                    CREATE TABLE orders_1 as
+                        SELECT orderkey as %s,
+                            custkey,
+                            orderstatus,
+                            totalprice,
+                            orderdate,
+                            orderpriority,
+                            clerk,
+                            shippriority,
+                            comment
+                        FROM orders
+                    """, validColumnName), "VALUES 15000");
+
+            Session session = Session.builder(getSession())
+                    .setCatalogSessionProperty("oracle", JOIN_PUSHDOWN_ENABLED, "true")
+                    .build();
+            assertQuery(session,
+                    format("""
+                            SELECT c.custkey, o.%s, n.nationkey
+                                 FROM orders_1 o JOIN customer c ON c.custkey = o.custkey
+                                 JOIN nation n ON c.nationkey = n.nationkey
+                            """, validColumnName),
+                    """
+                                SELECT c.custkey, o.orderkey, n.nationkey
+                                FROM orders o JOIN customer c ON c.custkey = o.custkey
+                                JOIN nation n ON c.nationkey = n.nationkey
+                            """);
+        }
+        finally {
+            tryCleanupTemporaryTable();
+        }
+    }
+
+    private void tryCleanupTemporaryTable()
+    {
+        try {
+            assertUpdate("DROP TABLE orders_1");
+        }
+        catch (Exception ex) {
+            ex.printStackTrace();
+        }
+    }
```

Comment:
```suggestion
    @Test
    public void testPushdownJoinWithLongNameSucceeds()
    {
        String maximumLengthColumnIdentifier = "z".repeat(MAX_CHARS_COLUMN_ALIAS);
        try (TestTable table = new TestTable(getQueryRunner()::execute, "long_identifier", "(%s bigint)".formatted(maximumLengthColumnIdentifier))) {
            assertThat(query(joinPushdownEnabled(getSession()), """
                SELECT r.name, t.%s, n.name
                FROM %s t JOIN region r ON r.regionkey = t.%s
                JOIN nation n ON r.regionkey = n.regionkey""".formatted(maximumLengthColumnIdentifier, table.getName(), maximumLengthColumnIdentifier)))
                    .isFullyPushedDown();
        }
    }

```

This seems to be sufficient to test.
Try to avoid creating data in tests since it can be slow, specially if using larger tables like orders. nation and region are best if your test doesn't care about size of tables.

Note: we need to verify that the query was indeed pushed down because otherwise it's possible that the connector would generate a plain SELECT with no synthetic ids and hence our test will pass.

Joins cannot always be pushed down even if join pushdown is enabled (e.g. if the join condition or some other operation in the query prevents pushdown).

Reply:
Also while this test by itself is sufficient to prove that the aliasing logic works it doesn't prove that we chose an optimal value for the max identifier length.

IMO we should have another test to verify the length we chose:

- Try create a column with length = max_length - must succeed
- Try create a column with length = max_length + 1 - must fail in expected manner

the reason why i believe this is useful is because for example we can use 30 as the max globally but it un-necessarily loses information in connectors which can actually support longer identifiers making reading explain plans harder than needed. Do you agree?

Reply:
> we can use 30 as the max globally but it un-necessarily loses information in connectors which can actually support longer identifiers making reading explain plans harder than needed

If that is the case, we can try to leave the old 'formatter', in case it's a non-Oracle implementation. Then the code will run exactly as was (possibly with a bug) for any jdbc that is non-Oracle. Not sure whether this is better (because we have the old logic that does not truncate, so it is easier to trace/debug), or worse (because there might still be a bug).

Reply:
Since we know we'll follow up on this I don't think either choice makes a difference. End result after the follow up will be that each JDBC connector will use *some* formatter.